packages/xml-views-validation/src/validators/elements/unknown-tag-name.ts

```diff
@@ -0,0 +1,199 @@
+import { UnknownTagNameIssue } from "../../../api";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { XMLElement, XMLToken } from "@xml-tools/ast";
+import {
+  getUI5ClassByXMLElement,
+  ui5NodeToFQN,
+  getUI5NodeFromXMLElementNamespace,
+  getUI5AggregationByXMLElement,
+} from "@ui5-language-assistant/logic-utils";
+import {
+  getMessage,
+  UNKNOWN_CLASS_IN_NS,
+  UNKNOWN_CLASS_WITHOUT_NS,
+  UNKNOWN_AGGREGATION_IN_CLASS,
+  UNKNOWN_TAG_NAME_IN_CLASS,
+} from "../../utils/messages";
+
+export function validateUnknownTagName(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): UnknownTagNameIssue[] {
+  // Can't give an error if there is no position
+  if (xmlElement.syntax.openName == undefined || xmlElement.name === null) {
+    return [];
+  }
+
+  // To avoid false positives, we only validate tags with recognized namespaces (from the model)
+  // or without a namespace (and for tags without a namespace, we only validate if
+  // they cannot be user-defined classes or aggregations).
+
+  // Get the namespace
+  if (xmlElement.ns !== undefined) {
+    // The cast is necessary because typescript doesn't recognize the conditions above as type guards
+    return validateTagWithNamespace(xmlElement as NamedXMLElementWithNS, model);
+  } else {
+    // The cast is necessary because typescript doesn't recognize the condition above as a type guard
+    return validateTagWithoutNamespace(xmlElement as NamedXMLElement, model);
+  }
+}
+
+type NamedXMLElement = XMLElement & {
+  name: string;
+  syntax: { openName: XMLToken };
+};
+type NamedXMLElementWithNS = NamedXMLElement & { ns: string };
+
+function validateTagWithNamespace(
+  xmlElement: NamedXMLElementWithNS,
+  model: UI5SemanticModel
+): UnknownTagNameIssue[] {
+  const issueDefaults = {
+    kind: "UnknownTagName" as "UnknownTagName",
+    offsetRange: {
+      start: xmlElement.syntax.openName.startOffset,
+      end: xmlElement.syntax.openName.endOffset,
+    },
+    severity: "error" as "error",
+  };
+
+  // To avoid false positives, we assume unrecognized namespaces are user-defined so we don't validate tags in them
```

Comment:
this comment should be above line 68 or line 69, closer to the condition or the empty return